src/v/cluster_link/link.cc

```diff
@@ -85,8 +85,7 @@ link::link(
   , _config(std::move(config))
   , _cluster_connection(std::move(cluster_connection))
   , _replication_mgr(
-      // todo: fix me
-      ss::default_scheduling_group(),
+      _manager->scheduling_group(),
```

Comment:
I wonder if we should plumb this sg into partition_replicator which do all the heavy lifting, currently this is only used. Currently this is only used for the work queue that drives meta operations.